flink-connectors/flink-connector-hive/src/main/java/org/apache/flink/connectors/hive/HiveTableInputFormat.java

```diff
@@ -206,6 +212,9 @@ public void close() throws IOException {
 
 	protected void fetchNext() throws IOException {
 		hasNext = this.recordReader.next(key, value);
+		if (limit > 0 && currentReadCount >= limit) {
```

Comment:
Put this check before reading from `recordReader ` so that we don't actually read any data once limit is reached?